### ISSUE TITLE
MenuFlyoutItem icon's foreground doesn't change back to resting color

### ISSUE THREAD

**Issue by eduardobragaxz** [microsoft/microsoft-ui-xaml#9690]
### Describe the bug

If you change the foreground of a MenuFlyoutItem and then hover over it, the color won't return for the icon once the pointer goes away.

### Steps to reproduce the bug

Create a project with a MenuFlyoutItem and change it's foreground.

### NuGet package version

WinUI 3 - Windows App SDK 1.5.3: 1.5.240428000

### Windows version

Windows 11 (22H2): Build 22621

**Comment by codendone** [comment]
How are you changing the foreground on the MenuFlyoutItem? If you set the `Foreground` property directly on the element, this will override any values being set from style.

**Comment by eduardobragaxz** [comment]
Yeah I just change the Foreground on the MenuFlyoutItem. If I hover the mouse the text and icon will change to white/black, and when I move the mouse away only the text reverts to the color I set. The icon remains white/black.

**Comment by eduardobragaxz** [comment]
https://github.com/microsoft/microsoft-ui-xaml/assets/50815710/a6a1982c-794d-43b5-a732-c02bd8829b03

You can see the very fist AppBarButton that is blue works correctly, but the two MenuFlyoutItems don't.

**Comment by codendone** [comment]
Thanks for clarifying!

**Comment by kmahone** [comment]
I think this is a similar root cause to #7208.

In the default template, the TextBlock has `Foreground="{TemplateBinding Foreground}"`, however the ContentPresenter for the Icon has no Foreground set. So when transitioning away from the PointerOver state, the icon foreground does not revert back to what it was previously.

**Comment by niels9001** [comment]
This issue has been fixed in 1.8-stable
